arangod/Replication2/StateMachines/Document/DocumentCore.cpp

```diff
@@ -47,30 +47,32 @@ DocumentCore::DocumentCore(
 
   // TODO this is currently still required. once the snapshot transfer contains
   //  a list of shards that exist for this log, we can remove this.
-  _shardId = _params.shardId;
+  auto shardId = _params.shardId;
   auto shardResult = _shardHandler->createLocalShard(
-      _shardId, _params.collectionId, collectionProperties);
+      shardId, _params.collectionId, collectionProperties);
   TRI_ASSERT(shardResult.ok()) << "Shard creation failed for replicated state"
                                << _gid << ": " << shardResult;
 
+  auto shardProperties = ShardProperties{
+      .shardId = shardId,
+      .collectionId = _params.collectionId,
+      .properties = std::move(collectionProperties),
+  };
+  _shards.emplace(shardId, std::move(shardProperties));
   LOG_CTX("b7e0d", TRACE, this->loggerContext)
-      << "Created shard " << _shardId << " for replicated state " << _gid;
+      << "Created shard " << shardId << " for replicated state " << _gid;
 }
 
-auto DocumentCore::getShardId() -> ShardID const& { return _shardId; }
+auto DocumentCore::getShardId() -> ShardID const& {
+  return _shards.begin()->first;
+}
```

Comment:
delete this function?

Reply:
We will, but for now we still need it. This depends on the snapshot transfer PR to be merged first.

Reply:
Not yet, unfortunately. We'll wait for #18109 to be merged.